packages/react-query/src/internals/getQueryKey.ts

```diff
@@ -1,11 +1,61 @@
+import {
+  AnyMutationProcedure,
+  AnyQueryProcedure,
+  AnyRouter,
+  inferProcedureInput,
+} from '@trpc/server';
+import { QueryType, getArrayQueryKey } from '../internals/getArrayQueryKey';
+import { DecorateProcedure, DecoratedProcedureRecord } from '../shared';
+
 /**
  * We treat `undefined` as an input the same as omitting an `input`
  * https://github.com/trpc/trpc/issues/2290
  */
-export function getQueryKey(
+export function getQueryKeyInternal(
   path: string,
   input: unknown,
 ): [string] | [string, unknown] {
   if (path.length) return input === undefined ? [path] : [path, input];
   return [] as unknown as [string];
 }
+
+/**
+ * Method to extract the query key for a procedure
+ * @param type - defaults to `any`
+ * @link https://trpc.io/docs/useContext#-the-function-i-want-isnt-here
+ */
+export function getQueryKey<
+  TProcedureOrRouter extends
+    | AnyQueryProcedure
+    | AnyMutationProcedure
+    | AnyRouter,
+  TPath extends string,
+  TFlags,
+>(
+  ..._params: TProcedureOrRouter extends AnyQueryProcedure
+    ? [
+        procedureOrRouter: DecorateProcedure<TProcedureOrRouter, TFlags, TPath>,
+        ..._params: inferProcedureInput<TProcedureOrRouter> extends undefined
+          ? []
+          : [input: inferProcedureInput<TProcedureOrRouter>, type?: QueryType],
```

Comment:
is this correct? i should be able to do `getQueryKey(myProc, undefined, 'infinite'` for example

Reply:
turns out that did work

![CleanShot 2023-01-21 at 00 31 33](https://user-images.githubusercontent.com/51714798/213823167-5cdefeca-5095-447d-af4b-f97812efcb21.png)